api/chat.go

```diff
@@ -45,61 +45,124 @@ func configGinChatRouter(router *gin.RouterGroup) {
 		if tumLiveContext.User == nil {
 			return
 		}
-		var chat ChatReq
-		if err := json.Unmarshal(msg, &chat); err != nil {
-			return
-		}
-		if !tumLiveContext.Course.ChatEnabled {
-			return
-		}
-		uname := tumLiveContext.User.Name
-		if chat.Anonymous && tumLiveContext.Course.AnonymousChatEnabled {
-			uname = "Anonymous"
-		}
-		replyTo := sql.NullInt64{}
-		if chat.ReplyTo == 0 {
-			replyTo.Int64 = 0
-			replyTo.Valid = false
-		} else {
-			replyTo.Int64 = chat.ReplyTo
-			replyTo.Valid = true
-		}
-		chatForDb := model.Chat{
-			UserID:   strconv.Itoa(int(tumLiveContext.User.ID)),
-			UserName: uname,
-			Message:  chat.Msg,
-			StreamID: tumLiveContext.Stream.ID,
-			Admin:    tumLiveContext.User.ID == tumLiveContext.Course.UserID,
-			ReplyTo:  replyTo,
-		}
-		err := dao.AddMessage(&chatForDb)
+		var req wsReq
+		err := json.Unmarshal(msg, &req)
 		if err != nil {
-			if errors.Is(err, model.ErrCooledDown) {
-				sendServerMessage("You are sending messages too fast. Please wait a bit.", TypeServerErr, s)
-			}
+			log.Warn(err)
```

Comment:
Maybe we should send a more verbose sentry exception here or even abort with an HTTP status code?

Reply:
Warn should be a good error level here, as users can potentially send whatever invalid data they want. I made it more verbose though :+1:

Reply:
Okay, perfect :+1: